certbot/ocsp.py

```diff
@@ -1,34 +1,45 @@
 """Tools for checking certificate revocation."""
 import logging
 import re
-
 from subprocess import Popen, PIPE
 
+try:
+    from cryptography.x509 import ocsp  # pylint: disable=import-error
+except ImportError:  # pragma: no cover
+    ocsp = None  # type: ignore
+from cryptography import x509
+from cryptography.hazmat.backends import default_backend
+from cryptography.hazmat.primitives import serialization
+from cryptography.hazmat.primitives.hashes import SHA1
+import requests
+
 from certbot import errors
 from certbot import util
 
 logger = logging.getLogger(__name__)
 
+
 class RevocationChecker(object):
-    "This class figures out OCSP checking on this system, and performs it."
+    """This class figures out OCSP checking on this system, and performs it."""
 
-    def __init__(self):
+    def __init__(self, enforce_openssl_binary_usage=False):
         self.broken = False
-
-        if not util.exe_exists("openssl"):
-            logger.info("openssl not installed, can't check revocation")
-            self.broken = True
-            return
-
-       # New versions of openssl want -header var=val, old ones want -header var val
-        test_host_format = Popen(["openssl", "ocsp", "-header", "var", "val"],
-                                 stdout=PIPE, stderr=PIPE, universal_newlines=True)
-        _out, err = test_host_format.communicate()
-        if "Missing =" in err:
-            self.host_args = lambda host: ["Host=" + host]
-        else:
-            self.host_args = lambda host: ["Host", host]
-
+        self.use_openssl_binary = not ocsp or enforce_openssl_binary_usage
```

Comment:
Can I suggest `enforce_openssl_binary_usage or ocsp is None` here as potentially clearer for a reader of this code?

Reply:
Indeed, clearer to read the positive assertions before the negative ones.